pkg/storage/pebbleiter/crdb_test_on.go

```diff
@@ -96,63 +102,98 @@ func (i *assertionIter) LazyValue() pebble.LazyValue {
 	return i.Iterator.LazyValue()
 }
 
+func (i *assertionIter) RangeBounds() ([]byte, []byte) {
+	if !i.Valid() {
+		panic(errors.AssertionFailedf("RangeBounds() called on !Valid() pebble.Iterator"))
+	}
+	if _, hasRange := i.Iterator.HasPointAndRange(); !hasRange {
+		panic(errors.AssertionFailedf("RangeBounds() called on pebble.Iterator without range keys"))
+	}
+	// See maybeSaveAndMangleRangeKeyBufs for where these are saved.
+	j := i.rangeKeyBufs.idx
+	return i.rangeKeyBufs.start[j], i.rangeKeyBufs.end[j]
+}
+
+func (i *assertionIter) RangeKeys() []pebble.RangeKeyData {
+	if !i.Valid() {
+		panic(errors.AssertionFailedf("RangeKeys() called on !Valid() pebble.Iterator"))
```

Comment:
Why isn't this valid? We document that at least `RangeBounds()` returns nothing when there are  no range keys, and I'd expect `RangeKeys()` to do the same. Is there a compelling reason not to allow this?